web/lattice_main/templatetags/lattice_main_tags.py

```diff
@@ -0,0 +1,75 @@
+from django import template
+# import sys
+# sys.path.append("..")
+from ..styles_map import styles_map
+
+register = template.Library()
+
+@register.filter
+def get_len(obj):
+    return len(obj)
+
+@register.filter
+def enum(obj):
+    enumerated = enumerate(obj)
+    return enumerated
+
+@register.filter
+def cell_patch_type(cell):
+    patch_type = cell.patch_type
+    return patch_type
+
+@register.filter
+def PatchType_Ancilla(patches):
+    pta = patches.PatchType.Ancilla
+    return pta
+
+@register.filter
+def styles_map_cell_patch_type(cell):
+    """ """
+    style = styles_map[cell.patch_type]
+    return style
+
+@register.filter
+def styles_map_orientation(orientation):
+    style = styles_map[orientation]
+    return style
+    
+@register.filter
+def styles_map_edge_color(edge_color):
+    style = styles_map[edge_color]
+    return style
+
+@register.filter
+def styles_map_edge_type(edge_type):
+    style = styles_map[edge_type]
+    return style
+
+@register.filter
+def cell_edges_items(cell):
+    items = cell.edges.items()
+    return items
+
+@register.filter
+def styles_map_edge_color_edge_type(edge_type):
+    """styles_map['edge_color'][edge_type]"""
+    style = styles_map['edge_color'][edge_type]
+    return style
+
+@register.filter
+def cell_activity(cell):
+    return cell.activity
+
+@register.filter
+def styles_map_activity_color_cell_activity_activity_type(cell):
+    style = styles_map['activity_color'][cell.activity.activity_type]
+    return style
+
+@register.filter
+def cell_text(cell):
+    text = cell.text
+    return text
+
```

Comment:
I see what you mean by it was tedious to translate the complicated template code from mako to django

Reply:
Since we already have this file, that for a large part is just mapping styles map stuff, perhaps the styles map could be coded in here from now on?

Reply:
Yea that styles map idea was really clashing with django's templates